### PR TITLE
implement socket counting for mac

### PR DESCRIPTION
**Pull request by dothebart**
since on mac the `jenkins` user doesn't have access to the systems global used socket state list, but to each individual processes socket list, this PR:
- moves the list of socket-states into a global static array
- for mac iterates over any process, filter them for `arangod`s
- fetch the socket statistics for all interesting processes and sumarizes them.
Since ArangoDs should be the main interesting socket creater & user during CI runs, this should give us the socket count information in valuable precision as the global socket count.

### CODE REVIEW

File: jenkins/helper/test_launch_controller.py [hunk 1]
```diff
@@ -497,24 +497,39 @@ def testing_runner(testing_instance, this, arangosh):
...
-                psutil.CONN_LAST_ACK,
-                psutil.CONN_LISTEN,
-                psutil.CONN_CLOSING
-        ]:
-            counter += 1
+    if IS_MAC:
+        # Mac would need root for all sockets, so we just look
+        # for arangods and their ports, which works without.
+        for proc in psutil.process_iter(['pid', 'name']):
+            if proc.name() != 'arangod':
```
Comment (KVS85) [inline jenkins/helper/test_launch_controller.py:521]: What about `arangosh`?

Comment (dothebart) [reply jenkins/helper/test_launch_controller.py:521]: hm, good point - could be underestimated.

**Comment by dothebart** [comment]
https://jenkins01.arangodb.biz/view/Mac/job/arangodb-ANY-mac-matrix.arm64/152/

**Comment by dothebart** [comment]
testrun doesn't show errors related to the PR.

File: jenkins/helper/test_launch_controller.py (hunk 1 above)
Comment (dothebart) [reply jenkins/helper/test_launch_controller.py]: implemented, testrun successfull.

**Review by fceller — APPROVED** [review]

**Pull request merged**